File: frontend/whitebear/src/app/models/product/product.ts
import { Category } from '../category/category';
import { Brand } from '../brand/brand';
import { Reaction } from '../reaction/reaction';

export interface Product {
    id: string;
    name: string;
    description: string;
    discount?: number;
    price: number; 
    beerType: number; 
    density: number;
    previewImg: string;
    raiting: number;
    category: Category;
    brand: Brand;
    reactions: Reaction[];
}

File: backend/src/Services/Catalog/Catalog.Api/Data/Entities/Product.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using WhiteBear.Services.Catalog.Api.Data.Entities.Abstraction;
using WhiteBear.Services.Catalog.Api.Enums;

namespace WhiteBear.Services.Catalog.Api.Data.Entities
{
    public sealed class Product : BaseEntity
    {
        [Required]
        [MinLength(2)]
        public string Name { get; set; }
        public string Description { get; set; }
        public double? Discount { get; set; }
        [Required]
        public decimal Price { get; set; }
        [Required]
        public EnumBeerTypes BeerType { get; set; }
        public double Density { get; set; } 
        public string ImageId { get; set; }
        public Image PreviewImg { get; set; }
        [Required]
        public string CategoryId { get; set; }
        public Category Category { get; set; }
        [Required]
        public string BrandId { get; set; }
        public Brand Brand { get; set; }
        public string ReactionId { get; set; }
        public ICollection<Reaction> Reactions { get; set; }
    }
}
